### PR TITLE
PMM-9630 Use status, status_code only for some versions.

### PR DESCRIPTION
**Pull request by JiriCtvrtka**
PMM-9630

Original PR: https://github.com/percona/pmm-agent/pull/327

Build: SUBMODULES-0

- [ ] Links to other linked pull requests (optional).

### CODE REVIEW

**Review by ShashankSinha252 — APPROVED** [review]

File: agent/agents/postgres/pgstatmonitor/pgstatmonitor.go [hunk 1]
```diff
@@ -72,6 +72,9 @@ const (
 	pgStatMonitorVersion10PG12
 	pgStatMonitorVersion10PG13
 	pgStatMonitorVersion10PG14
+	pgStatMonitorVersion11PG12
```
Comment (BupycHuk) [inline agent/agents/postgres/pgstatmonitor/pgstatmonitor.go:75]: new variables are not added to this function https://github.com/percona/pmm/pull/912/files#diff-ae12be77883993fc7d2b5ebc501e2fb0b0078ea1d4645c2f9327095ae0661d30R153

File: agent/agents/postgres/pgstatmonitor/stat_monitor_cache.go [hunk 2]
```diff
@@ -91,10 +91,11 @@ func (ssc *statMonitorCache) getStatMonitorExtended(ctx context.Context, q *refo
 
 	row, view := NewPgStatMonitorStructs(pgMonitorVersion)
 	conditions := "WHERE queryid IS NOT NULL AND query IS NOT NULL"
-	if pgMonitorVersion >= pgStatMonitorVersion09 {
-		// only pg_stat_monitor 0.9.0 and above supports state_code. It tells what is the query's current state.
+	if pgMonitorVersion >= pgStatMonitorVersion09 && pgMonitorVersion <= pgStatMonitorVersion11PG14 && prerelease != "" {
```
Comment (BupycHuk) [inline agent/agents/postgres/pgstatmonitor/stat_monitor_cache.go:94]: I thought that we decided to have a check like `version >= 0.9 and version <= 2.0`

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: agent/agents/postgres/pgstatmonitor/pgstatmonitor.go (hunk 1 above)
Comment (JiriCtvrtka) [reply agent/agents/postgres/pgstatmonitor/pgstatmonitor.go:75]: Added.

File: agent/agents/postgres/pgstatmonitor/stat_monitor_cache.go (hunk 2 above)
Comment (JiriCtvrtka) [reply agent/agents/postgres/pgstatmonitor/stat_monitor_cache.go]: Changed.

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**